docs/architecture/adr/0025-ts-deprecate-enums.md

```diff
@@ -0,0 +1,90 @@
+---
+adr: "0025"
+status: Proposed
+date: 2025-05-30
+tags: [clients, typescript]
+---
+
+# 0025 - Deprecate TypeScript Enum Types
+
+<AdrTable frontMatter={frontMatter}></AdrTable>
+
+## Context and Problem Statement
+
+TypeScript experts have long discouraged the use of enums. There are a variety of cited reasons.
+Among them:
+
+- Enums that are not `const` substantially increase output size.
+- Numeric enums implicitly cast from `number` to the enum type, which allows invalid enum values to
+  be transmitted to functions.
+- String enums are named types that require a member is used, which is inconsistent with the
+  behavior of numeric enums.
+
+These inconsistencies cause increased complexity from guard statements in the best of cases. In the
+worst cases, their limitations may be unknown, and thus unguarded. Reducing or eliminating
+
+### Detection
+
+Typescript deprecation can be linted using a fairly short ESlint plugin. The code has [already been
+contributed to main][no-enum-lint] as a suggestion. The same PR adds `FIXME` comments for each team
+to address.
+
+### Replacement code
+
+In most cases, enums are unnecessary. A readonly (`as const`) object coupled with a type alias
+avoids both code generation and type inconsistencies.
+
+```ts
+export const CipherType = {
+  Login: 1,
+  SecureNote: 2,
+  Card: 3,
+  Identity: 4,
+  SshKey: 5,
+} as const;
+
+export type CipherType = (typeof CipherType)[keyof typeof CipherType];
+
+// Can be imported together
+import { CipherType } from "./cipher-type";
+
+// Used as a type
+function doSomething(type: CipherType) {}
+
+// And used as a value (just like a regular `enum`)
+doSomething(CipherType.Card);
+```
+
+## Considered Options
+
+- **Allow enums, but advise against them** - This is the current state of affairs. With this option,
+  teams MUST address the readmes, but MAY address them by disabling the lint.
```

Comment:
```suggestion
  teams **must** address the readmes, but _may_ address them by disabling the lint.
```
We do use bold and italic on this site.

Reply:
I was aiming for RFC definitions for those words. I guess we don't do that?